treeherder/model/derived/jobs.py

```diff
@@ -1773,7 +1773,8 @@ def _load_jobs(self, job_placeholders, job_guid_list, retry_job_guids):
                 job_id_lookup[retry_guid] = retry_job
                 # for the intermediate representation (see below), we only
                 # want to modify the job who was retriggered
-                del jobs_to_update[retry_guid_root]
+                if jobs_to_update.get(retry_guid_root):
```

Comment:
Since we're not using the return value, rather than `.get()` we could use `if retry_guid_root in jobs_to_update:`, or else just try-except.

Or alternatively in one line `jobs_to_update.pop(retry_guid_root)` (although that's potentially slightly less clear that it's doing a delete).